frame/transaction-payment/src/lib.rs

```diff
@@ -229,6 +265,32 @@ decl_module! {
 					<T as frame_system::Trait>::MaximumBlockWeight::get().try_into().unwrap()
 				).unwrap(),
 			);
+
+			// This is the minimum value of the multiplier. Make sure that if we collapse to this
+			// value, we can recover with a reasonable amount of traffic. For this test we assert
+			// that if we collapse to minimum, the trend will be positive with a weight value
+			// which is 1% more than the target.
+			let min_value = T::FeeMultiplierUpdate::min();
+			let mut target =
+				T::FeeMultiplierUpdate::target() *
+				(T::AvailableBlockRatio::get() * T::MaximumBlockWeight::get());
+
+			// add 1 percent;
+			let addition = target / 100;
+			if addition == 0 {
```

Comment:
Do we really need this? If yes, your trait should return an `Option`.

Reply:
Return an option from the `integrity_test()`?  Not sure what you mean. 

The comment below explains this. In some crates we call `construct_runtime!` with dummy values and then `target` is `0` and this test would fail, thus, early return. 

Ideally, I prefer this to never allow any `construct_runtime!` in a real runtime to expand with values that cause `target == 0`. The solution would be to add a token to `construct_runtime!` to disable integration tests. Then we can use that relaxed version in test mocks.